### PR TITLE
Jains18

### PR DESCRIPTION
**Pull request by sonika-nih**
CHARMS Fanconi Survey verifying the Medical History Continued module from the portal and adding SMS (ResearchActivities and SubjectFlags)module.

### CODE REVIEW

**Review by alenaminsk — APPROVED** [review]

**Review by katm007 — APPROVED** [review]
Approved

File: src/test/java/ServiceNow/CHARMS/Playwright/Runners/FanconiStudy/Run_CHARMS_FanconiSurvey_Regression_Test.java [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 import io.cucumber.testng.CucumberOptions;
 
-/* we would have to come back and update this runners class at a later time.*/
 @CucumberOptions(plugin = {"html:target/html-reports/cucumber-default-report.html", "json:target/cucumber.json",
         "rerun:target/failed.txt",
         "pretty", "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"},
         features = "src/test/java/ServiceNow/CHARMS/FanconiSurvey/Playwright/Features",
         glue = {"ServiceNow.CHARMS.FanconiSurvey.Playwright.Steps","Hooks"},
-        tags = "@Regression",
+        tags = "@In-Progress",
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Playwright/Runners/FanconiStudy/Run_CHARMS_FanconiSurvey_Regression_Test.java:11]: This is a regression runner and should only have @Regression tag

File: src/test/java/ServiceNow/CHARMS/Playwright/Runners/SMS/Run_CHARMS_SMS_Regression_Test.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(plugin = {"html:target/html-reports/cucumber-default-report.html", "json:target/cucumber.json",
+        "rerun:target/failed.txt",
+        "pretty", "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"},
+        features = "src/test/java/ServiceNow/CHARMS/FanconiSurvey/Playwright/Features",
+        glue = {"ServiceNow.CHARMS.FanconiSurvey.Playwright.Steps","Hooks"},
+        tags = "@In-Progress",
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Playwright/Runners/SMS/Run_CHARMS_SMS_Regression_Test.java:11]: This is a regression runner and should only have @regression tag

File: src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java [hunk 3]
```diff
@@ -1,12 +1,13 @@
-package ServiceNow.CHARMS.FanconiSurvey.Playwright.StepsImplementation;
-
-import ServiceNow.CHARMS.FanconiSurvey.Playwright.Pages.FanconiSurveyLoginPage;
+package ServiceNow.CHARMS.Playwright.StepsImplementation.FanconiStudy;
+import ServiceNow.CHARMS.Playwright.Pages.FanconiStudy.FanconiSurveyLoginPage;
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java:2]: please an empty line between package and imports

File: src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 import com.microsoft.playwright.Locator;
 import com.microsoft.playwright.Page;
 import com.microsoft.playwright.options.AriaRole;
 import com.nci.automation.utils.CucumberLogUtils;
 import com.nci.automation.web.PlaywrightUtils;
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java:8]: please delete the empty line

File: src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java [hunk 5]
```diff
@@ -1408,5 +1409,1055 @@ public static void medicalHistoryContinuedSectionScenario1() {
...
+        assertThat(page.getByRole(AriaRole.OPTION, new Page.GetByRoleOptions().setName("Cancer prevention/Prophylaxis"))).isVisible();
+        assertThat(page.getByRole(AriaRole.OPTION, new Page.GetByRoleOptions().setName("Other (please specify)"))).isVisible();
+        assertThat(page.getByRole(AriaRole.OPTION, new Page.GetByRoleOptions().setName("Unknown/Unsure"))).isVisible();
+        page.getByRole(AriaRole.OPTION, new Page.GetByRoleOptions().setName("Cysts")).click();
+        page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName(FanconiSurveyLoginPage.SUBMIT)).click();
     }
-}
\ No newline at end of file
+
+    public static void surgicalAndOtherMedicalHistorySectionScenario1(){
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java:1794]: Please add a Javadoc for this method

File: src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java [hunk 6]
```diff
@@ -1408,5 +1409,1055 @@ public static void medicalHistoryContinuedSectionScenario1() {
...
+        page.locator("a").filter(new Locator.FilterOptions().setHasText("-- None --")).click();
+        assertThat(page.getByRole(AriaRole.OPTION, new Page.GetByRoleOptions().setName("to 2"))).isVisible();
+        assertThat(page.getByRole(AriaRole.OPTION, new Page.GetByRoleOptions().setName("to 5"))).isVisible();
+        assertThat(page.getByRole(AriaRole.OPTION, new Page.GetByRoleOptions().setName("to 9"))).isVisible();
+        assertThat(page.getByRole(AriaRole.OPTION, new Page.GetByRoleOptions().setName("or more"))).isVisible();
+        assertThat(page.getByRole(AriaRole.OPTION, new Page.GetByRoleOptions().setName("Unknown/Unsure"))).isVisible();
+        page.getByRole(AriaRole.OPTION, new Page.GetByRoleOptions().setName("to 2")).click();
+    }
+}
+
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java:2463]: please the empty line from the bottom of the page

File: src/test/java/ServiceNow/CHARMS/SMS/Playwright/Runners/Run_CHARMS_SMS_Regression_Test.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(plugin = {"html:target/html-reports/cucumber-default-report.html", "json:target/cucumber.json",
+        "rerun:target/failed.txt",
+        "pretty", "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"},
+        features = "src/test/java/ServiceNow/CHARMS/SMS/Playwright/Features",
+        glue = {"ServiceNow.CHARMS.SMS.Playwright.Steps","Hooks"},
+        tags = "@In-Progress",
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/SMS/Playwright/Runners/Run_CHARMS_SMS_Regression_Test.java:11]: This is a regression runner and should only have @regression tag

**Review by hamidnesar — CHANGES_REQUESTED** [review]
Hi, please review the comments

File: src/test/java/ServiceNow/CHARMS/SMS/Playwright/Runners/Run_CHARMS_SMS_Regression_Test.java (hunk 7 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/SMS/Playwright/Runners/Run_CHARMS_SMS_Regression_Test.java:11]: Done

File: src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java (hunk 6 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java:2463]: Done

File: src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java (hunk 3 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Playwright/StepsImplementation/FanconiStudy/FanconiSurveyStepsImpl.java:2]: Done

**Review by hamidnesar — APPROVED** [review]
Looks good!

**Pull request merged**